languages/ruby/spec/oso/polar/polar_spec.rb

```diff
@@ -476,13 +476,24 @@ def x
                      def my_method(one, two, three:, four:)
                        [one, two, three, four]
                      end
+
+                    def self.tv
+                      Thread.current["foo"]
+                    end
                    end)
         subject.register_class(TestClass)
       end
 
       it 'accepts positional and keyword args' do
         expect(qvar(subject, 'x = (new TestClass()).my_method(1, 2, four: 4, three: 3)', 'x')).to eq([[1, 2, 3, 4]])
       end
+
+      it 'has access to the thread context' do
+        Thread.current["foo"] = "bar"
+        # you need to use load string here or it might not
```

Comment:
I don't fully get this, but if you use `expect(qvar(...` like the other test cases the bad behaviour doesn't get triggered.

Reply:
I'm guessing if you return the enumerator in the test, then Ruby figures out it doesn't need to create a new fiber to evaluate the query, so it all happens on the right thread.